### PR TITLE
Set a minimum version for pyyaml

### LINKED ISSUE #8543: Set minimum pyyaml version
- Currently, in `setup.py`, there is not a minimum pyyaml version set
  - https://github.com/dask/dask/blob/3c46e89aea2af010e69049cd638094fea2ddd576/setup.py#L37
- It would be ideal if dask set the minimum require version

### PR DESCRIPTION
**Pull request by gsheni**
- [x] Closes #8543
- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by jakirkham** [comment]
ok to test

**Comment by gsheni** [comment]
@jrbourbeau this PR is ready for review.

File: continuous_integration/hdfs/install.sh [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
 set -xe
 
-docker exec hdfs conda install -y -q dask pyarrow">=0.14.0,!=2.0.0" fsspec pyyaml openjdk -c conda-forge
+docker exec hdfs conda install -y -q dask pyarrow">=0.14.0,!=2.0.0" fsspec pyyaml">=5.3.1" openjdk -c conda-forge
```
Comment (gsheni) [inline continuous_integration/hdfs/install.sh:4]: @jrbourbeau @jsignell I wasn't sure if this needed to be changed or not

**Comment by gsheni** [comment]
The tests pass locally for me (macOS). The windows-tests seem also be failing on `main`.

File: continuous_integration/hdfs/install.sh (hunk 1 above)
Comment (jsignell) [reply continuous_integration/hdfs/install.sh:4]: I can't see how it would hurt :)

**Review by jsignell — APPROVED** [review]
This looks good to me! Thanks @gsheni for taking on this work :)

**Review by jrbourbeau — COMMENTED** [review]
Yeah, thanks for adding this @gsheni

**Pull request merged**